### PR TITLE
Update torch.autograd.graph logging to not print out grad_output

### PR DESCRIPTION
**Pull request by soulitzer**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #116710
* __->__ #116523

Instead of printing the tensor's data print the dtype and shape metadata of the tensor.
```
Executing: <VarMeanBackward0 object at 0x1352d0e20> with grad_outputs: [None,f32[]]
```
This is important in order to avoid doing a cuda sync and also useful to reduce verbosity.

### CODE REVIEW

File: torch/autograd/graph.py [hunk 1]
```diff
@@ -659,7 +659,7 @@ def iter_graph(roots):
 
     def prehook(grad_output):
         node = torch._C._current_autograd_node()
-        log_str = f"Executing: {node} with grad_output: {grad_output}"
+        log_str = f"Executing: {node}"
```
Comment (albanD) [inline torch/autograd/graph.py:662]: If this is too verbose, feel free to only print size to share some information.

**Review by albanD — APPROVED** [review]
SGTM

File: torch/autograd/graph.py [hunk 2]
```diff
@@ -657,9 +657,18 @@ def iter_graph(roots):
...
             yield node
 
-    def prehook(grad_output):
+    def fmt(t):
+        # Avoid circular import
+        from torch.testing._internal.common_utils import dtype_abbrs
+
+        if t is None:
+            return "None"
+        return f"{dtype_abbrs[t.dtype]}[{', '.join(map(str, t.shape))}]"
```
Comment (albanD) [inline torch/autograd/graph.py:666]: Is that a usual format used in other places?

Comment (soulitzer) [reply torch/autograd/graph.py:666]: Yeah, fx and logging tensor also use this format

**Review by albanD — COMMENTED** [review]
Change sounds good but needs rebase!

**Review by albanD — APPROVED** [review]

**Pull request closed without merging**